### ISSUE TITLE
isErrorFromAlias not finding errors when uri have undefined params

### ISSUE THREAD

**Issue by Zehir** [ecyrbe/zodios#402]
Hello,

TLDR; `isErrorFromAlias` is checking the `isErrorFromPath` does.

I have an issue where, now I can fix easily but what really hard to debug.

I was using `isErrorFromAlias` but was returning false and I was not able to figure out why. I tried using a z.any() schema, an 'default' status but none of them worked.

After copying the code of `isErrorFromAlias` in my project and add many many console.log I was able to find the issue.

It's related to this code :
https://github.com/ecyrbe/zodios/blob/e8232fcb5490263dd24b4effde622245843073f7/src/utils.ts#L111-L125

Even it's looking for the endpoint using the alias it's still checking the uri.

The issue is in the `pathMatchesUrl` method ;
https://github.com/ecyrbe/zodios/blob/e8232fcb5490263dd24b4effde622245843073f7/src/utils.ts#L127-L130

But in my case the endpoint URI is `/api/:apiKey/config/whitelist/:oldApiKey` and the url of the Axios response is `/api/SOME_KEY/config/whitelist/`

The regex is resolved to `/api/([^/]+)/config/whitelist/([^/]+)` but that don't match the response url because oldApiKey was undefined.

The easy fix it to make sure the oldApiKey is defined.

My question, if it's looking with the alias why it's still checking the uri ?

And a side question, does the alias need to be uniques in the api ?

Thanks

**Comment by ecyrbe** [comment]
it's checking the uri, because that's all the error that got rejected has to provide to match the call. The alias alone can't allow to match the error uri.

In v11, i'am now evaluating to remove this entirely and make the call return the object error, like this :

```ts
const { data: user, error } = client.getUser();
```
instead of :  

```ts
try {
 const user = client.getUser();
 } catch (e) {
   if(isErrorFromAlias('getUser')) {
   ...
   }
 }
 ```
 
 But the decision to do this breaking change is not yet settled

**Comment by Zehir** [comment]
Yes but you could use the alias to get the endpoint (like you did) and only the status for the error ;

```typescript
   const endpoint = findEndpointByAlias(api, alias); 
  
   return (endpoint && 
     err.config && 
     endpoint.method === err.config.method)
     ? findEndpointErrors(endpoint, err) 
     : undefined; 
```
Also you probably should add ( ) around the condition, it's more readable and typescript don't complain that way

**Comment by thelinuxlich** [comment]
I want to use https://github.com/causaly/zod-validation-error for better errors, what would be nice is to check the error like this:

```ts
if(err instanceof ZodiosError) {
   console.log(fromZodError(err.zodError).toString())
}
```
